test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java

```diff
@@ -91,6 +91,12 @@ public boolean clusterHasFeature(String featureId, boolean any) {
         }
 
         if (hasFeatureMetadata()) {
+            if (isRestApiCompatibilityTest()) {
```

Comment:
This is required because the compatibility tests take the test definitions from the earlier version, which contains conditions on various features. Given there's other checks that features are not removed, the only way this can happen here is if the feature has been bona-fide removed as part of moving past a compatibility barrier.

Reply:
I vaguely remember we also had `yamlRestTestV7CompatTest`. That probably has to turn into it's v8 counterpart? Any implications there?

Reply:
I can't find any reference to that in the codebase. I would still expect it to set the system parameter though (also used by `ReproduceInfoPrinter`)

Reply:
That was a bit of an open general question not specific to the compat test property, very unsure myself ... I had a brief look for yamlRestTestV7CompatTest, but couldn't find it anymore. Honestly also not sure anymore what was covered by that compatibility suite. I was wondering if a v8 version of that is needed, and if so, if we have to keep a version of the specs that preserves the previous features (though, in that case - even later - that version could just be restored from whatever existed before this PR get's merged)